GH Actions: run DevTools tests on Windows

rodrigoprimo opened the pull request:
This PR changes both the test.yml and the quicktest.yml workflows to run the tests on Windows as well. Before, the tests were executed only on Linux.

Installing xmllint on Windows using Chocolatey is a bit slow. We can look into ways to make it faster if needed.

Another thing to consider is whether we need to run all the test variations on Windows, as done in this PR. If not, this should speed up a bit the total time for all the checks to complete.

rodrigoprimo commented:
I'm not totally sure, but I believe the required checks are defined in the GitHub repository settings, and those will need to be updated if this PR is approved as is, as the name of the checks changes to include the OS name.

jrfnl commented on .github/workflows/quicktest.yml:
```diff
@@ -17,28 +17,38 @@ concurrency:
         php: ['5.4', 'latest']
 
-    name: "QTest + Lint: PHP ${{ matrix.php }}"
+    name: "QTest + Lint: PHP ${{ matrix.php }} - OS ${{ matrix.os }}"
```
The official image names are long and make the titles less easy to scan. This should improve their readability.

```suggestion
    name: "QTest + Lint: PHP ${{ matrix.php }} (${{ matrix.os == 'windows-latest' && 'Win' || 'Linux' }})"
```

rodrigoprimo replied:
That is a fair point. I will update both files once we decide which jobs should run against Windows.

jrfnl reviewed (commented):
Thanks for setting this up @rodrigoprimo !

Two questions:
1. Why was this pulled against `stable` ? If didn't look at in detail when we discussed yesterday, but `stable` at this time contains no tests for which the OS makes a difference.
2. Why are you adding runs against Windows for the `quicktest-phpcs` and `test-phpcs` jobs ?

rodrigoprimo commented:
Thanks for checking this PR, @jrfnl!

> Why was this pulled against stable ? I didn't look at it in detail when we discussed yesterday, but stable at this time contains no tests for which the OS makes a difference.

Maybe I misunderstood our conversation yesterday, but I thought you asked me to pull this PR against `stable`, so I didn't think much about it. That being said, `stable` contains the tests for PHPCSDebug that have special handling for different line endings. Example: https://github.com/PHPCSStandards/PHPCSDevTools/blob/stable/PHPCSDebug/Tests/Debug/TokenListJsTest.php#L163-L166. Considering this, maybe it makes sense to keep this PR against `stable`?

> Why are you adding runs against Windows for the quicktest-phpcs and test-phpcs jobs ?

Because they contain the PHPCSDebug tests with special handling for different line endings as I detailed above. But maybe for some reason that I'm missing, you don't think it is necessary to run quicktest-phpcs and test-phpcs jobs against Windows just for this reason?

jrfnl commented:
> That being said, `stable` contains the tests for PHPCSDebug that have special handling for different line endings. Considering this, maybe it makes sense to keep this PR against `stable`?

No, the use of `PHP_EOL` alone in the sniff is not a reason for running the tests against Windows. In that case, we'd be testing PHP, not our own code.

rodrigoprimo commented:
I went ahead and updated the original commit to run only `test-php` and `quicktest-php` on Windows. I also added in a separate (to be squashed before merge) your suggestion to make the job name shorter.

I created a separate branch on my fork to verify that the quicktest is working as expected: https://github.com/rodrigoprimo/PHPCSDevTools/actions/runs/15047189089

This PR is ready for another review. Thanks.

jrfnl commented:
Happy to merge this PR to `develop`. The thing to have a think about is whether this will need to be backported to `stable` if the docs tooling will be pulled to `stable` and if so, whether that will cause conflicts. If so, it might be better to pull it as part of the docs tooling PR, to be merged up into `develop` later.

Let me know what your thoughts are about this.
